tests/e2e/specs/factory/Factory.spec.ts

```diff
@@ -131,6 +131,8 @@ suite(
 			const scmView: NewScmView = new NewScmView();
 			await driverHelper.waitVisibility(webCheCodeLocators.ScmView.inputField);
 			[scmProvider, ...rest] = await scmView.getProviders();
+			await projectAndFileTests.manageWorkspaceTrust(scmProvider);
```

Comment:
IMHO, we can make the logic of script more obvious and avoid second execution of `[scmProvider, ...rest] = await scmView.getProviders()` when extract `if (scmProvider === undefined) {` verification from `manageWorkspaceTrust()` to test script level.

Reply:
Fixed